dumpling/export/dump.go

```diff
@@ -1427,7 +1427,7 @@ func resolveAutoConsistency(d *Dumper) error {
 		err = FlushTableWithReadLock(d.tctx, conn)
```

Comment:
Thanks for your contribution! I think the problem is dumpling will be blocked at this line, which means it infinitely waits for MySQL FTWRL. Your changes at line 1430 will not help.

I think you can test at your own local environment for this issue. Start a testing MySQL instance. Connect to it and start a transaction that you will not commit / rollback it during the test. Then start dumpling. After the fix dumpling can fallback to LOCK TABLES.